### PR TITLE
dev: prefer podman to docker

### PR DESCRIPTION
**Pull request by mastercactapus**
**Description:**
This PR changes the default container tool from `docker` to `podman` that can be used as a drop-in replacement. This is to keep the default toolset accessible with recent licensing changes with the docker tool.

If `podman` is unavailable make targets will fallback to `docker` automatically resulting in no required changes for current developers. Docker can be forced by setting the tool at the end of the make command like this: `make start CONTAINER_TOOL=docker`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
-.PHONY: stop start build-docker lint tools regendb resetdb
+.PHONY: stop start lint tools regendb resetdb
 .PHONY: smoketest generate check all test test-long install install-race
 .PHONY: cy-wide cy-mobile cy-wide-prod cy-mobile-prod cypress postgres
 .PHONY: config.json.bak jest new-migration check-all cy-wide-prod-run cy-mobile-prod-run
-.PHONY: docker-goalert docker-all-in-one release force-yarn
+.PHONY: goaler-container demo-container release force-yarn
```
Comment (m17ch) [inline Makefile:5]: `goaler-container` typo?

Comment (mastercactapus) [reply Makefile]: fixed :) ty

**Review by m17ch — APPROVED** [review]
LGTM
Functionally validated after podman installed -- everything appeared to be working as desired.

**Review by arurao — APPROVED** [review]
LGTM :tada:

**Pull request merged**